### PR TITLE
chore: Add stopwatch to log execution time by process name

### LINKED ISSUE #11068: [Feature]: Implement stopwatch to log execution time
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Git APIs are slow today and these also involves lot of operation:
- Hydration and rehydration from and to mongo DB
- Copying the serialised objects file system
- Working with local as well as remote git APIs

This makes it a little harder to point out which task is consuming the bandwidth.

### Why should this be worked on?

This will help us to analyze which part of the code executions needs more attention for optimizing time complexities

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

Git APIs are slow today and these also involves a lot of operation:
- Hydration and rehydration from and to Mongo DB
- Copying the serialized objects file system
- Working with local as well as remote git APIs

Current PR simply logs the execution time in milliseconds for the above-mentioned tasks, which can be utilized to optimize the implementation

Fixes #11068 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Manual test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by abhvsn** [comment]
@AnaghHegde Please let me know if we want to log the execution time in other places as well if I missed any

**Review by AnaghHegde — COMMENTED** [review]
Should we add the timer for importApplicationIntoOrg method as well? As it is commonly used across lot of git operations.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ImportExportApplicationServiceCEImpl.java [hunk 1]
```diff
@@ -534,6 +538,10 @@
             6. Extract and save pages in the application
             7. Extract and save actions in the application
          */
+
+        // Start the stopwatch to log the execution time
+        Stopwatch processStopwatch = new Stopwatch("Import application");
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ImportExportApplicationServiceCEImpl.java:543]: @AnaghHegde this is already present for import as well as export

File: app/server/appsmith-git/src/main/java/com/appsmith/git/service/GitExecutorImpl.java [hunk 2]
```diff
@@ -375,6 +395,7 @@ private Path createRepoPath(Path suffix) {
                                                  String privateKey,
                                                  String publicKey,
                                                  Boolean refreshBranches) {
+        Stopwatch processStopwatch = new Stopwatch("JGIT listBranches");
```
Comment (AnaghHegde) [inline app/server/appsmith-git/src/main/java/com/appsmith/git/service/GitExecutorImpl.java:398]: Can we include the flag as well to identify the pruneBranch vs normal list branch in the name?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ImportExportApplicationServiceCEImpl.java (hunk 1 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ImportExportApplicationServiceCEImpl.java:543]: My bad, i missed it somehow.

**Review by AnaghHegde — APPROVED** [review]
LGTM

**Pull request merged**